dsref/version_info.go

```diff
@@ -7,15 +7,18 @@ import (
 	"github.com/qri-io/dataset"
 )
 
-// VersionInfo is the way that everything talks about information about datasets in the cases
-// that an entire Dataset is not being returned. Used as an output value for many methods, such
-// as List, and Log.
+// VersionInfo is an aggregation of fields from a dataset version for caching &
+// listing purposes. VersionInfos are typically used when showing a list of
+// datasets or a list of dataset versions ("qri list" and "qri log").
+//
+// VersionInfos can also describe dataset versions that are being created or
+// failed to create. In these cases the calculated VersionInfo.Path value must
+// always equal the empty string.
```

Comment:
The addition of this caveat is kinda irritating. I'm of the mind that we'd clarify that subsytems like dscache and  the refstore only hold `VersionInfo`s that represent HEAD (the latest valid commit).